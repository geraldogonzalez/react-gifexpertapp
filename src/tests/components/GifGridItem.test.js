import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas sobre <GifGridItem />', () => {

    const title = 'Un título cualquiera';
    const url = 'https://una+URL+cualquiera';
    const wrapper = shallow( 
        <GifGridItem 
            title={ title }
            url={ url }
        /> 
    );
    
    test('Debe mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe mostrar el texto en el párrafo correctamente.', () => {
        
        const p = wrapper.find('p').text();

        expect( p.trim() ).toBe( title );

    });
    
    test('Debe mostrar los atributos de la imagen correctamente.', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });
    
    test('Debe mostrar la clase "animate__fadeIn".', () => {
        
        const div = wrapper.find('div');
        expect( div.prop('className') ).toContain('animate__fadeIn')

        // const { className } = div.props();
        // expect( className.includes('animate__fadeIn') ).toBe(true)

    });


    

})
