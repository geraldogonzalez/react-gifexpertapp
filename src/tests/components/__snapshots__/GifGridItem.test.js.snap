// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título cualquiera"
    src="https://una+URL+cualquiera"
    title="Un título cualquiera"
  />
  <p>
     
    Un título cualquiera
     
  </p>
</div>
`;
